src/ophyd_async/epics/signal/_p4p_table_model.py

```diff
@@ -0,0 +1,60 @@
+import numpy as np
+from pydantic import BaseModel, ConfigDict, model_validator
+
+
+class PvaTable(BaseModel):
+    """An abstraction of a PVA Table of str to numpy array."""
+
+    model_config = ConfigDict(validate_assignment=True, strict=False)
+
+    @classmethod
+    def row(cls, sub_cls, **kwargs) -> "PvaTable":
+        arrayified_kwargs = {
+            field_name: np.concatenate(
+                (
+                    (default_arr := field_value.default_factory()),
+                    np.array([kwargs[field_name]], dtype=default_arr.dtype),
+                )
+            )
+            for field_name, field_value in sub_cls.model_fields.items()
+        }
+        return sub_cls(**arrayified_kwargs)
+
+    def __add__(self, right: "PvaTable") -> "PvaTable":
+        """Concatenate the arrays in field values."""
+
+        assert isinstance(right, type(self)), (
+            f"{right} is not a `PvaTable`, or is not the same "
+            f"type of `PvaTable` as {self}."
+        )
+
+        return type(self)(
+            **{
+                field_name: np.concatenate(
+                    (getattr(self, field_name), getattr(right, field_name))
+                )
+                for field_name in self.model_fields
+            }
+        )
+
+    @model_validator(mode="after")
+    def validate_arrays(self) -> "PvaTable":
+        first_length = len(next(iter(self))[1])
+        assert all(
+            len(field_value) == first_length for _, field_value in self
+        ), "Rows should all be of equal size."
+
+        assert 0 <= first_length < 4096, f"Length {first_length} not in range."
```

Comment:
This is only a limit on the PandA SeqTable, no other tables have a limit of 4096 rows

Reply:
Tbh I was hoping I could specify this in the `Field` with `lt=4096` I'll try and do this again.

Reply:
Hmmm this doesn't work and I don't really know why :thinking: We could add an abstract property called `max_length` in `SeqTable` which would be used in the check in `Table`?